stl/inc/iterator

```diff
@@ -155,8 +171,8 @@ public:
     }
 
 protected:
-    _Container* container; // pointer to container
-    typename _Container::iterator iter; // iterator into container
+    _Container* container              = nullptr;
+    typename _Container::iterator iter = typename _Container::iterator();

```

Comment:
Even if you use copy-init, we should use empty braces for consistency with our new improved conventions. (I'm not sure if we can use direct-list-init; can we?)

Reply:
I'm mindlessly following the `type name = type();` style in the working draft which is intended to defend against Tim Song types (aggregates with a subobject that has an explicit default constructor). That style is unnecessary for "new" iterators since we fixed the `default_initializable` concept to require `T{}` to be valid. _`Cpp17DefaultConstructible`_ already required `T{}` for _`Cpp17ForwardIterator`_, I'll pretend that the handwavy front matter in [insert.iterators] actually requires `Container` to meet the container requirements and have a _`Cpp17ForwardIterator`_.

TLDR: Yes, we can direct-list-init or copy-list-init.